filebeat/inputsource/common/streaming/listener.go

```diff
@@ -62,6 +64,23 @@ var (
 	}
 )
 
+// Unpack unpacks the FramingType string value.
+func (f *FramingType) Unpack(value string) error {
+	ft, ok := framingTypes[value]
+	if !ok {
+		availableTypes := make([]string, len(framingTypes))
+		i := 0
+		for t := range framingTypes {
+			availableTypes[i] = t
+			i++
+		}
+		return fmt.Errorf("invalid framing type '%s', supported types: %v", value, availableTypes)
+
```

Comment:
`framingTypes` is never mutated AFAICS, so maybe the format string here could be constructed in an `init`?

Also I recall seeing recently that the indexed slice from map construction used here is no longer more efficient than the append-into-correctly-capped construction. If an `init` is used even that care is probably not worth the readability cost.

Reply:
Done (This code was only moved up from the bottom to be next to the type declaration.)